syntax = "proto3";

package smokers;

// Zutaten IDs
enum Ingredient {
    UNKNOWN = 0;  // Default, muss 0 sein
    PAPER = 1;
    TOBACCO = 2;
    MATCH = 3;
}

// Nachrichten
message IngredientMessage {
    Ingredient ingredient = 1;
}

message IngredientList {
    repeated Ingredient ingredients = 1;
}

message Empty {}

message NotifyRequest {}
message NotifyResponse {
    string message = 1;
}

message ContinueRequest {}
message ContinueResponse {}

message RegisterRequest {
    Ingredient ingredient = 1;
    string address = 2;
}

message RegisterResponse {
    bool success = 1;
}

// Service für den Tisch (Table)
service TableService {
    rpc PutIngredient(IngredientMessage) returns (Empty);
    rpc TakeIngredient(IngredientMessage) returns (Empty);
    rpc CheckIngredients(Empty) returns (IngredientList);
    rpc RegisterSmoker(RegisterRequest) returns (RegisterResponse);
}



// Service für den Dealer
service DealerService {
    rpc ContinueRound(ContinueRequest) returns (ContinueResponse);
}

// Service für den Smoker
service SmokerService {
    rpc Notify(NotifyRequest) returns (NotifyResponse);
}
